### PR TITLE
fix: unhelpful error message when `base` is missing from YAML

### LINKED ISSUE #997: Unhelpful errors for missing package information keys
## Situation

When Rockcraft validates the project file, the errors it provide are inconsistent depending on the key.

For example, if `name` is omitted, it prints:

```
Bad rockcraft.yaml content:
- field 'name' required in top-level configuration
```

But if `base` is absent, it prints:

```
rockcraft internal error: AttributeError("'NoneType' object has no attribute 'count'")
``` 

It's impossible to identify the issue from that message.

Related to #182.

## Request

For the mandatory top-level/metadata/package information keys, we should be providing better validation messages. Those are the most critical for publication and least likely to have actual validation errors.

### To Reproduce

Pack the provided project file.

### Environment

Ubuntu 24.04.

### rockcraft.yaml

```yaml
name: test
# see https://documentation.ubuntu.com/rockcraft/en/1.14.0/explanation/bases/
# for more information about bases and using 'bare' bases for chiselled rocks
version: '0.1' # just for humans. Semantic versioning is recommended
# base: ubuntu@24.04
summary: Single-line elevator pitch for your amazing rock # 79 char long summary
description: |
    This is empty's description. You have a paragraph or two to tell the
    most important story about it. Keep it under 100 words though,
    we live in tweetspace and your description wants to look good in the
    container registries out there.
platforms: # the platforms this rock should be built on and run on
    amd64:

parts:
    my-part:
        plugin: nil
```

### Relevant log output

```shell
rockcraft internal error: AttributeError("'NoneType' object has no attribute 'count'")
```

### PR DESCRIPTION
**Pull request by steinbro**
Fixes #997

- [ ] Have you followed the guidelines for contributing?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `make lint && make test`?

---

### CODE REVIEW

File: rockcraft/pebble.py [hunk 1]
```diff
@@ -260,7 +260,8 @@ def add_pebble_part(project: dict[str, Any]) -> None:
     :raises CraftValidationError: If `project` already contains a "pebble" part,
       and said part's contents are different from the contents of the part we add.
     """
-    build_base = project.get("build-base") or project["base"]
+    # at least base should be required, but YAML has not been validated yet
+    build_base = project.get("build-base") or project.get("base", "")
```
[the remark this answers is not in the document]
Comment (steinbro) [reply rockcraft/pebble.py]: I think the right thing to do here is just bail from the method; a missing base means we shouldn't even try to figure out which pebble part to add. I've tweaked the logic accordingly.

[the remark this answers is not in the document]
Comment (steinbro) [reply rockcraft/pebble.py]: I added a test case to confirm that `add_pebble_part` doesn't choke on an empty/invalid base. This revealed another confusing error message now properly fixed by the revised logic, when a YAML file is missing a base but has a predefined pebble part (`Cannot change the default "pebble" part`).

File: tests/unit/services/test_project.py [hunk 2]
```diff
@@ -92,6 +92,9 @@ def test_add_pebble_part_noble(
     [
         ("bare", "devel"),
         ("ubuntu@25.10", "devel"),
+        # base is not allowed to be empty, but checking nonetheless since
+        # pebble part is added pre-YAML validation
```
Comment (lengau) [inline tests/unit/services/test_project.py]: Could you add a negative test for this as well (that if there's no pebble part and both base and build_base are None, it doesn't add the pebble part)?

File: rockcraft/pebble.py [hunk 3]
```diff
@@ -260,7 +260,12 @@ def add_pebble_part(project: dict[str, Any]) -> None:
     :raises CraftValidationError: If `project` already contains a "pebble" part,
       and said part's contents are different from the contents of the part we add.
     """
-    build_base = project.get("build-base") or project["base"]
+    # at least base should be required, but YAML has not been validated yet
+    build_base = project.get("build-base") or project.get("base")
+    if build_base is None:
+        # don't try to add the relevant pebble part if base is missing
+        # (friendly error message will be presented later from pydantic)
```
[the remark this answers is not in the document]
Comment (steinbro) [reply rockcraft/pebble.py]: Added tests to explicitly check that pydantic errors are shown even after extensions are applied.

**Review by tigarmo — APPROVED** [review]

**Review by lengau — APPROVED** [review]
